src/hnsw/insert.c

```diff
@@ -156,6 +155,8 @@ bool ldb_aminsert(Relation         index,
     hdr_page = GenericXLogRegisterBuffer(state, hdr_buf, LDB_GENERIC_XLOG_DELTA_IMAGE);
     hdr = (HnswIndexHeaderPage *)PageGetContents(hdr_page);
 
+    HEADER_FOR_EXTERNAL_RETRIEVER = *hdr;
```

Comment:
during every insert we're reading header page, so makes sense to update retriever's header now